packages/react/src/transactions/useFollow.ts

```diff
@@ -1,126 +1,109 @@
+        followerAddress: follower.ownedBy,
+        followerProfileId: follower.id,
       };
     case FollowPolicyType.ANYONE:
-      return baseRequest;
-    case null:
-      throw new Error('Can`t follow with unsupported follow fee module');
+      return {
+        kind: TransactionKind.FOLLOW_PROFILES,
+        profileId: followee.id,
+        followerAddress: follower.ownedBy,
+      };
     case FollowPolicyType.NO_ONE:
-      throw new Error('Can`t follow with revert follow fee module');
+      throw new InvariantError(
+        `The profile is configured so that nobody can follow it. Check 'profile.followPolicy.type' beforehand.`,
+      );
     case FollowPolicyType.UNKNOWN:
-      throw new Error('Can`t follow with unknown follow module');
+      throw new InvariantError(
+        `The profile is configured with an unknown follow module. Check 'profile.followPolicy.type' beforehand.`,
+      );
   }
 }
 
 export type UseFollowArgs = {
-  profile: ProfileFragment;
+  followee: ProfileFragment;
+  follower: ProfileOwnedByMeFragment;
 };
 
-export function useFollow({ profile }: UseFollowArgs) {
-  const [error, setError] = useState<
-    | InsufficientAllowanceError
-    | InsufficientFundsError
-    | PendingSigningRequestError
-    | UserRejectedError
-    | WalletConnectionError
-    | null
-  >(null);
-  const [isPending, setIsPending] = useState<boolean>(false);
-  const follow = useFollowController();
-  const activeWallet = useActiveWalletVar();
-  const { data: activeProfile } = useActiveProfile();
-
-  return {
-    follow: async () => {
-      try {
-        invariant(activeWallet && activeProfile, 'You must be logged in to follow a profile');
-        invariant(
-          profile.followPolicy?.type !== FollowPolicyType.UNKNOWN,
-          'Unsupported follow module',
-        );
-        setIsPending(true);
-        setError(null);
+export type FollowOperation = Operation<
+  void,
+  | InsufficientAllowanceError
+  | InsufficientFundsError
+  | PendingSigningRequestError
+  | PrematureFollowError
+  | UserRejectedError
+  | WalletConnectionError
+>;
 
-        const request = createFollowProfilesFlowRequest(profile, activeWallet, activeProfile);
+export function useFollow({ followee, follower }: UseFollowArgs): FollowOperation {
+  const follow = useFollowController();
 
-        const result = await follow({
-          profileId: request.profileId,
-          followerAddress: request.followerAddress,
-          followerProfileId: request.followerProfileId,
-          fee: request.fee,
-          kind: TransactionKind.FOLLOW_PROFILES,
-        });
+  const hasPendingUnfollowTx = useHasPendingTransaction(
+    isUnfollowTransactionFor({ profileId: followee.id }),
+  );
 
-        if (result.isFailure()) {
-          setError(result.error);
-        }
-      } finally {
-        setIsPending(false);
+  return useOperation(
+    async (): PromiseResult<
+      void,
+      | InsufficientAllowanceError
```

Comment:
nit: could be worth conflating the errors here and on line 70 into a single type to avoid duplication

e.g
```typescript
type FollowErrors =
  | InsufficientAllowanceError
  | InsufficientFundsError
  | PendingSigningRequestError
  | PrematureFollowError
  | UserRejectedError
  | WalletConnectionError;
  ```

Reply:
I gave it a try but seem it hides the union in IDE suggestions at the call site:
<img width="451" alt="image" src="https://user-images.githubusercontent.com/3353250/218736001-0aa91be5-75b6-4e42-b9be-afd8aeef268c.png">

which IMO is counterproductive. We are exposing failure modes so to increase awareness of them.

I understand the repetition but don't have a better suggestion than keep the union.